### PR TITLE
Spendless V.1.1.0

### PR DESCRIPTION
**Pull request by nickemma**
# Hi there 👋🏻

**Thanks for taking the time to review the project. Here are some things important to note about it:**

### Video Demo

_You can find a video demo of the project [here](https://www.loom.com/share/186e5fb4fef24b2d9120088f07f702fc)_

## Now into the project...

**This project was done following git-flow, for which we have different pull requests, each of them with specific information about each feature. You can refer to them below**

- ✅ https://github.com/NickEmma/Spendless-App/pull/1. Here was the basic project configuration
- ✅ https://github.com/NickEmma/Spendless-App/pull/2. Here the default authentication failure was defined to redirect to the splash screen instead of the login screen
- ✅https://github.com/NickEmma/Spendless-App/pull/3. The categories page was defined for the authenticated user
- ✅https://github.com/NickEmma/Spendless-App/pull/4. The transactions page was defined for the authenticated user
- ✅ https://github.com/NickEmma/Spendless-App/pull/5. The deployment to Heroku was done here, as well as the setup of Sendgrid
- ✅ The project follows the style guidelines
- ✅ The project was developed following the TDD approach. Thus, almost every feature is tested

# Final notes 📔

_Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review._

### CODE REVIEW

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @NickEmma , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no **live version** link on your README file

- [ ] Please deployed your project and add the deploy link to your project README file.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by nickemma** [comment]
@CollinsTatang I deployed it but the styles did not apply on production. that is why I did not add it there

File: README.md [hunk 1]
```diff
@@ -1 +1,127 @@
...
+
+- [Ruby](https://www.ruby-lang.org/en/)
+- [Postgresql](https://www.postgresql.org/)
+- [Node.js](https://nodejs.org/en/)
+- [Yarn](https://yarnpkg.com/)
+- [Rails](https://rubyonrails.org/)
+
+## Click here for the live version
+
+- [live demo](https://mysterious-retreat-34191.herokuapp.com/users/sign_in)
```
Comment (OLIPLICHE) [inline README.md]: - **Your App is well deployed, but we cannot see the home and details(The styling aspect) as expected, Kindly check it again.** 
-
Capture
- 

- **NB: I opened the link in two different machines to cross-check**.

File: app/views/categories/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    </div>
+
+    <div class="field">
+      <%= f.label :icon, class: 'hidden' %>
+      <%= f.text_field :icon, autofocus: true, class: "w-96 border border-slate-300 hover:border-indigo-300 p-4", placeholder: 'Icon (Please add a proper URL)' %>
+    </div>
+
+    <div class="actions mt-5">
+      <%= f.submit "Save", class: "px-6 py-2 text-white bg-green-500" %>
+    </div>
```
Comment (OLIPLICHE) [inline app/views/categories/new.html.erb:16]: - Nice Job 🥇 🥇🥇🥇

Capture

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+RSpec.describe User, type: :model do
+  subject do
+    User.new(name: 'Rida Arif', email: 'example@mail.com', password: 'password')
+  end
+  before { subject.save }
+
+  it 'name should not be nil' do
+    subject.name = nil
+    expect(subject).to_not be_valid
```
Comment (OLIPLICHE) [inline spec/models/user_spec.rb:11]: - Nice done 👍 
image

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @NickEmma., 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow was correctly used👏👏, then merged to GitHub flow.

## Required Changes ♻️
- Congratulations on a nice video presentation, kindly record again and add a piece of the code you find interesting or are proud of making.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nickemma** [comment]
Hi @OLIPLICHE I have made the changes thank you

File: README.md (hunk 1 above)
Comment (nickemma) [reply README.md]: @OLIPLICHE Thank you. Heroku is not displaying the stylings

File: app/views/categories/new.html.erb (hunk 2 above)
Comment (nickemma) [reply app/views/categories/new.html.erb:16]: @OLIPLICHE thank you for the compliments.

File: spec/models/user_spec.rb (hunk 3 above)
Comment (nickemma) [reply spec/models/user_spec.rb:11]: @OLIPLICHE thanks for your review

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @NickEmma👋, 

### You did a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**